modules/cnf-topology-aware-lifecycle-manager-precache-user-spec-images.adoc

```diff
@@ -0,0 +1,259 @@
+// Module included in the following assemblies:
+//
+// * scalability_and_performance/ztp_far_edge/ztp-talm-updating-managed-policies.adoc
+
+:_content-type: PROCEDURE
+[id="talm-prechache-user-specified-images_{context}"]
+= Pre-caching user specified images with {cgu-operator} on {sno} clusters
+
+In earlier {product-title} versions, you could pre-cache {product-title} workloads on {sno} clusters in a cluster with limited bandwidth.
+You can also pre-cache application-specific workload images on {sno} clusters before upgrading your applications.
+
+With the `PreCachingConfig` custom resource (CR) referenced in the `ClusterGroupUpgrade` CR, you can specify the configuration options for the pre-caching jobs.
+
+.Example ClusterGroupUpgrade CR with PreCachingConfig CR reference
+[source,yaml]
+----
+apiVersion: ran.openshift.io/v1alpha1
+kind: ClusterGroupUpgrade
+metadata:
+  name: cgu
+spec:
+  preCaching: true <1>
+  preCachingConfigRef:
+    name: exampleconfig
+    namespace: exampleconfig-ns
+----
+<1> The `preCaching` field set to `true` enables the pre-caching job.
+
+[NOTE]
+====
+All fields in the `PreCachingConfig` CR are optional.
+====
+
+.Example PreCachingConfig CR
+[source,yaml]
+----
+apiVersion: ran.openshift.io/v1alpha1
+kind: PreCachingConfig
+metadata:
+  name: exampleconfig
+spec:
+  overrides: <1>
+    platformImage: quay.io/openshift-release-dev/ocp-release@sha256:3d5800990dee7cd4727d3fe238a97e2d2976d3808fc925ada29c559a47e2e1ef
+    operatorsIndexes:
+      - registry.example.com:5000/custom-redhat-operators:1.0.0
+    operatorsPackagesAndChannels:
+      - local-storage-operator: stable
+      - ptp-operator: stable
+      - sriov-network-operator: stable
+  spaceRequired: 30 Gi <2>
+  excludePrecachePatterns: <3>
+    - aws
+    - vsphere
+  additionalImages: <4>
+    - quay.io/exampleconfig/application1@sha256:3d5800990dee7cd4727d3fe238a97e2d2976d3808fc925ada29c559a47e2e1ef
+    - quay.io/exampleconfig/application2@sha256:3d5800123dee7cd4727d3fe238a97e2d2976d3808fc925ada29c559a47adfaef
+    - quay.io/exampleconfig/applicationN@sha256:4fe1334adfafadsf987123adfffdaf1243340adfafdedga0991234afdadfsa09
+----
+<1> The `platformImage`, `operatorsIndexes`, and the `operatorsPackagesAndChannels` fields can be configured to override the default {cgu-operator}-derived values. These fields are automatically populated from the policies of the managed clusters if they are not specified.
+<2> Specifies the minimum required disk space on the cluster in Gigabytes. If unspecified, {cgu-operator} defines a default value for {product-title} images. The disk space field must include an integer value and the storage unit. For example: `40 GiB`, `200 MB`, `1 TiB`.
+<3> Specifies the list of images that can be excluded from the cluster version update.
```

Comment:
Possible correction- excludePrecachePatterns specifies which images to exclude from precaching based on image name matching.